Toned down interstitials for top-level blocking.

mkarolin opened the pull request:
Also:
* switched to Chromium's information (i) icon instead of the red exclamation point.
* changes the "Proceed" button to be the default (highlighted).

Fixes brave/brave-browser#15433

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

simonhong reviewed (approved):
++

simonhong commented:
I think we need sub issue from https://github.com/brave/brave-browser/issues/15189 for this fix.

fmarier commented on components/resources/domain_block_strings.grdp:
```diff
@@ -4,13 +4,13 @@
   </message>
   <message name="IDS_DOMAIN_BLOCK_EXPLANATION" desc="Second paragraph of domain block interstitial page.">
-    This site may be fraudulent, bounce you between various sites to set tracking cookies, or perform other privacy invasive actions.
+    This page may be trying to link some of your personal data with a cookie before taking you to your intended destination. You can choose to allow this potential privacy leak and proceed to the next page, or you can block this tracking attempt by going back to the previous page.
```
Let's use @karenkliu 's wording (see the screenshots instead of the GitHub comments) instead of what I proposed.

fmarier replied:
Sorry, I should have commented to that effect on the issue.

mkarolin replied:
@fmarier the heading as well?

fmarier reviewed (changes requested):
Would it be easy to change the default button too? In the screenshots, the "Proceed" button is the one that's highlighted.

mkarolin commented:
Switched "Proceed" button to be the default.
Switched to the text from the screenshot in brave/brave-browser#15189
Created a sub-issue that his PR fixes (brave/brave-browser#15433)

simonhong reviewed (approved):
++

fmarier reviewed (approved):
Thanks @mkarolin !

kjozwiak commented:
Verification **PASSED** on `Win 10 x64` using the following build:

```
Brave | 1.25.42 Chromium: 90.0.4430.85 (Official Build) nightly (64-bit)
-- | --
Revision | 5bc145d831c180d9ff94f29a0d7a2e1cbd30ef36-refs/branch-heads/4430@{#1311}
OS | Windows 10 OS Version 2009 (Build 19042.928)
```

* ensured that the the current interstitials page matches https://github.com/brave/brave-core/pull/8603#issuecomment-824483000 & https://github.com/brave/brave-browser/issues/15189#issue-852997454
* ensured that both `Proceed` & `Go back` buttons are working as expected
* ensured that `Don't warn me about this again` worked as expected

![Screenshot 2021-04-26 221757](https://user-images.githubusercontent.com/2602313/116175511-74f38900-a6de-11eb-8f7b-4bd7f6dcf974.png)
